src/__experimental__/components/search/search.component.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef, useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import invariant from 'invariant';
+import StyledSearch, { StyledSearchButton, StyledButtonIcon } from './search.style';
+import Icon from '../../../components/icon';
+import Textbox from '../textbox';
+import Button from '../../../components/button';
+
+const Search = ({
+  defaultValue, onChange, onClick, value, id, name, threshold, searchButton, placeholder, ...rest
+}) => {
+  const isControlled = value !== undefined;
+  const initialValue = isControlled ? value : defaultValue;
+  invariant(
+    typeof initialValue === 'string',
+    'This component has no initial value'
+  );
+  let inputRef = useRef(null);
+  const [searchValue, setSearchValue] = useState(initialValue);
+  const [isActive, setIsActive] = useState(inputRef.current === document.activeElement);
+  const [searchIsActive, setSearchIsActive] = useState(initialValue.length >= threshold);
+  const [iconType, setIconType] = useState(null);
+
+  const handleChange = (e) => {
+    if (onChange) {
+      onChange(e);
+    }
+    setSearchValue(e.target.value);
+  };
+
+  const handleOnFocus = () => {
+    setIsActive(true);
+  };
+
+  let buttonProps = {};
```

Comment:
```suggestion
  let buttonProps;
```

Reply:
Couldn't this potentially lead to spreading `{ ...undefined }` in `<Button>`? I also felt that the original way was more readable.